### PR TITLE
Feater/rewards csv impotr

### PR DESCRIPTION
**Pull request by mdobosz3**
Admin can add rewards via CSV file.
Rewards that already exist can be overwritten using the "slug" column in the reward database.
If the data format in the "slug" column is wrong or the reward to be overwritten does not exist, the entire operation will be rolled back  (transaction used).

### CODE REVIEW

File: app/models/reward.rb [hunk 1]
```diff
@@ -12,4 +12,19 @@ class Reward < ApplicationRecord
   has_many :categories, through: :category_rewards
 
   has_one_attached :photo
+
+  def self.import(file)
+    ActiveRecord::Base.transaction do
+      CSV.foreach(file.path, headers: true) do |row|
+        if row[3].nil?
```
Comment (oskarlakner) [inline app/models/reward.rb:19]: wdyt about using a multi-assignment here? That would increase the code readability :)
Screenshot 2022-05-04 at 15 09 09

File: app/models/reward.rb [hunk 2]
```diff
@@ -12,4 +12,19 @@ class Reward < ApplicationRecord
...
+          Reward.create_with(description: row[1], price: row[2]).find_or_create_by!(title: row[0], slug: row[0].parameterize)
```
Comment (oskarlakner) [inline app/models/reward.rb:20]: you should find a reward only by its slug. What if an admin updates the title of a reward? :)

**Review by oskarlakner — APPROVED** [review]
🍻

File: app/models/reward.rb (hunk 1 above)
Comment (mdobosz3) [reply app/models/reward.rb]: You're right, it will improve readability. I corrected it

File: app/models/reward.rb (hunk 2 above)
Comment (mdobosz3) [reply app/models/reward.rb]: I changed it :)

**Pull request merged**